backend/fregepoc/analyzers/core/typescript.py

```diff
@@ -0,0 +1,35 @@
+from typing import TypedDict
+
+import lizard
+
+from fregepoc.analyzers.core.base import AnalyzerFactory, BaseAnalyzer
+from fregepoc.repositories.constants import ProgrammingLanguages
+from fregepoc.repositories.utils.analyzers import repo_file_content
+
+
+class TypescriptFileAnalysisResult(TypedDict):
+    lines_of_code: int
+    token_count: int
+    average_lines_of_code: float
+    average_token_count: float
+    average_cyclomatic_complexity: float
+
+
+@AnalyzerFactory.register(ProgrammingLanguages.TYPESCRIPT)
+class JavascriptAnalyzer(BaseAnalyzer[TypescriptFileAnalysisResult]):
```

Comment:
I guess `TypescriptAnalyzer` could be a better name for the class :)

Reply:
Also, how is this different from JavaScript analyzer? Can't we use the same analyzer for both languages? If there are no differences, I don't see a point of having that class duplicated. Are there any TS-specific metrics? Could you add some for this class?

Reply:
I can't really think of any TS-specific metrics that wouldn't apply to JS, however lizard (which both TS and JS analyzers use) treats this languages differently, hence I belive it is problematic to use one analyzer for both JS and TS